targets/TARGET_TOSHIBA/TARGET_TMPM46B/trng_api.c

```diff
@@ -0,0 +1,88 @@
+/* mbed Microcontroller Library
+ * (C)Copyright TOSHIBA ELECTRONIC DEVICES & STORAGE CORPORATION 2017 All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed_error.h"
+#include "tmpm46b_esg.h"
+#include "trng_api.h"
+
+#define ESG_BUSY_SET                  (0x00000001U)
+
+static Result ESG_Config(void)
+{
+    Result ret = SUCCESS;
+    uint16_t Fintming = 560U;
+    // Confirm the ESG core stops
```

Comment:
Please use the error-by-default approach used in `tmpm46b_esg.c`, initializing `ret = ERROR` and clearing it to `ret = SUCCESS` in the unique successful code path. Further, please consider the return values of `ESG_SetLatchTiming` and `ESG_SetFintiming`, too.

Reply:
Initialized with ret = ERROR, and updated SUCCESS based on ESG_SetLatchTiming and ESG_SetFintiming return values.